src/ucp/core/ucp_context.c

```diff
@@ -576,6 +583,22 @@ static ucs_status_t ucp_fill_config(ucp_context_h context,
         ucs_warn("empty features set passed to ucp context create");
     }
 
+#if ENABLE_MT
+    context->mt_config.mt_required  = params->mt_workers_shared;
+#else
+    context->mt_config.mt_required  = 0;
+#endif
+    context->mt_config.use_mutex    = config->ctx.use_mt_mutex;
+
+    UCP_THREAD_LOCK_INIT(context->mt_config);
+
+    if (num_contexts == 0) {
+        mm_mt_config.mt_required = 1;
```

Comment:
Yes, we can do that. 
In that way, the if branch "if(context MT is turned on)" cannot be in the CS macro, but out of CS macro. Which means the lock code becomes "if (context MT is turned on) { CS_ENTER(contest->lock)}".